library/src/main/java/com/yayandroid/locationmanager/providers/locationprovider/DefaultLocationProvider.java

```diff
@@ -224,11 +227,9 @@ public void onLocationChanged(Location location) {
             getSourceProvider().getProviderSwitchTask().stop();
         }
 
-        getSourceProvider().removeLocationUpdates(this);
-
-        if (getConfiguration().keepTracking()) {
-            requestUpdateLocation(getConfiguration().defaultProviderConfiguration().requiredTimeInterval(),
-                  getConfiguration().defaultProviderConfiguration().requiredDistanceInterval(), false);
+        if (!getConfiguration().keepTracking()) {
```

Comment:
Instead of removing and registering new requests, only remove if keepTracking is false